src/Microsoft.ML.Core/Data/IEstimator.cs

```diff
@@ -237,7 +237,7 @@ public interface IEstimator<out TTransformer>
         /// <summary>
         /// Train and return a transformer.
         /// </summary>
-        TTransformer Fit(IDataView input);
+        TTransformer Fit(IDataView input, IDataView validationData = null, IPredictor initialPredictor = null);
```

Comment:
> IDataView validationData = null, IPredictor initialPredictor = null) [](start = 41, length = 69)

I am not certain I welcome this move. It seems like it is repeating the mistake of `IIncrementalValidatingTrainer`, which would be bad enough if it was just for trainers alone but it seems to be for absolutely every estimator.

Reply:
I agree

---
In reply to: [215803874](https://github.com/dotnet/machinelearning/pull/849#discussion_r215803874) [](ancestors = 215803874)